src/main/java/com/weirddev/testme/intellij/template/context/Field.java

```diff
@@ -42,21 +54,55 @@ public class Field {
     public Field(PsiField psiField, PsiClass srcClass, TypeDictionary typeDictionary, int maxRecursionDepth) {
         this.name = psiField.getName();
         type= buildType(psiField.getType(), typeDictionary, maxRecursionDepth);
+        this.isAnnotatedByDI = buildAnnotatedByDI(psiField);
```

Comment:
both these methods - buildAnnotationByDI & buildHasSetter - have nested loops and create collections. though the loop range would be small, there's a potential performance penalty. think it would be good to init these properties only if `typeDictionary.isTestSubject(srcClass)` as these new properties currently used only for test subject
similar check can be added to Type constructor as well